vlrouter.go

```diff
@@ -900,81 +969,100 @@ func (self *Vlrouter) initFgraph() error {
 3)  Proxy with interface mac is arp request is from remote endpoint
 4) Learn MAC,Port of the source if its not learnt and it is bgp peer endpoint
 */
-func (self *Vlrouter) processArp(pkt protocol.Ethernet, inPort uint32) {
-	log.Debugf("processing ARP packet on port %d", inPort)
+func (vl *Vlrouter) processArp(pkt protocol.Ethernet, inPort uint32) {
+	log.Infof("processing ARP packet on port %d", inPort)
 	switch t := pkt.Data.(type) {
 	case *protocol.ARP:
-		log.Debugf("ARP packet: %+v", *t)
+		log.Infof("ARP packet: %+v", *t)
 		var arpHdr protocol.ARP = *t
 		var srcMac net.HardwareAddr
 		var intf *net.Interface
 		var err error
 
-		self.agent.incrStats("ArpPktRcvd")
+		vl.agent.incrStats("ArpPktRcvd")
 
 		switch arpHdr.Operation {
 		case protocol.Type_Request:
-			self.agent.incrStats("ArpReqRcvd")
+			vl.agent.incrStats("ArpReqRcvd")
 
 			// Lookup the Dest IP in the endpoint table
-			endpoint := self.agent.getEndpointByIpVrf(arpHdr.IPDst, "default")
+			endpoint := vl.agent.getEndpointByIpVrf(arpHdr.IPDst, "default")
 			if endpoint == nil {
 				// Look for a service entry for the target IP
-				proxyMac := self.svcProxy.GetSvcProxyMAC(arpHdr.IPDst)
+				proxyMac := vl.svcProxy.GetSvcProxyMAC(arpHdr.IPDst)
 				if proxyMac == "" {
 					// If we dont know the IP address, dont send an ARP response
-					log.Debugf("Received ARP request for unknown IP: %v", arpHdr.IPDst)
-					self.agent.incrStats("ArpReqUnknownDest")
+					log.Infof("Received ARP request for unknown IP: %v", arpHdr.IPDst)
+					vl.agent.incrStats("ArpReqUnknownDest")
 					return
 				}
 				srcMac, _ = net.ParseMAC(proxyMac)
 			} else {
-				if endpoint.EndpointType == "internal" || endpoint.EndpointType == "internal-bgp" {
-					//srcMac, _ = net.ParseMAC(endpoint.MacAddrStr)
-					if self.agent.GetRouterInfo() != nil {
-						uplink := self.agent.GetRouterInfo().UplinkPort
-						intf, err = net.InterfaceByName(uplink.Name)
-						if err != nil {
-							log.Errorf("Error getting interface information. Err: %+v", err)
+				srcEp := vl.agent.getLocalEndpoint(inPort)
+				if vl.agent.isInternal(endpoint) || vl.agent.isInternalBgp(endpoint) {
+					if srcEp != nil && vl.agent.isInternal(srcEp) {
+						srcMac = vl.anycastMac
+					} else {
+						if vl.agent.GetRouterInfo() != nil {
+							uplink := vl.agent.GetRouterInfo().UplinkPort
+
+							if uplink != nil && len(uplink.MbrLinks) == 0 {
+								log.Errorf("Error getting interface information. Err: No member links present")
+								return
+							}
+
+							intf, err = net.InterfaceByName(uplink.MbrLinks[0].Name)
+							if err != nil {
+								log.Errorf("Error getting interface information. Err: %+v", err)
+								return
+							}
+						} else if vl.uplinkPortDb.Count() > 0 {
+							for ul := range vl.uplinkPortDb.IterBuffered() {
+								uplink := ul.Val.(*PortInfo)
+								if uplink != nil && len(uplink.MbrLinks) > 0 {
+									intf, err = net.InterfaceByName(uplink.MbrLinks[0].Name)
```

Comment:
Can you handle the case where we have more than one uplink? I understand that we allow only one uplink as of now, but it would be good to handle all the scenarios in cases like these. It would save a lot of debugging time when the support is added later.